pkg/collector/corechecks/cluster/ksm/kubernetes_state_aggregators.go

```diff
@@ -161,18 +168,34 @@ func (a *sumValuesAggregator) accumulate(metric ksmstore.DDMetric) {
 func (a *countObjectsAggregator) accumulate(metric ksmstore.DDMetric) {
 	var labelValues [maxNumberOfAllowedLabels]string
 
-	for i, allowedLabel := range a.allowedLabels {
-		if allowedLabel == "" {
+	for i := range a.allowedLabels {
+		if a.allowedLabels[i] == "" {
 			break
 		}
 
-		labelValues[i] = metric.Labels[allowedLabel]
+		labelValues[i] = metric.Labels[a.allowedLabels[i]]
 	}
 
 	a.accumulator[labelValues]++
 }
 
+func renameMig(resource string) string {
+	if strings.HasPrefix(resource, "nvidia_com_mig") {
+		return "gpu"
+	}
+	return resource
+}
+
 func (a *resourceAggregator) accumulate(metric ksmstore.DDMetric) {
+	resource := renameMig(metric.Labels["resource"])
+	if newName, ok := renamedResource[resource]; ok {
+		resource = newName
+	}
```

Comment:
Maybe we can move all the code related to resource renaming in a single function:
* move the `renamedResource` lookup inside the function
* rename the `renameMig` function to `renameResource`

```suggestion
var gpuResource = map[string]struct{}{
	"nvidia_com_gpu":     struct{}{},
	"amd_com_gpu":        struct{}{},
	"gpu_intel_com_i915": struct{}{},
}

func renameResource(resource string) string {
	if strings.HasPrefix(resource, "nvidia_com_mig") {
		return "gpu"
	}
	
	if _, found := renamedResource[resource]; found {
		return "gpu"
	}
	
	return resource
}

func (a *resourceAggregator) accumulate(metric ksmstore.DDMetric) {
	resource := renameResource(metric.Labels["resource"])
```